src/restorer/mnemonic-restorer.ts

```diff
@@ -22,36 +22,42 @@ function restorerFromEsplora<R extends MasterPublicKey>(
     gapLimit = 20,
   }) => {
     const restoreFunc = async function(
-      nextAddrFunc: (index: number) => string
-    ): Promise<number> {
+      nextAddrFunc: (index?: number) => string | undefined
+    ): Promise<number | undefined> {
       let counter = 0;
       let index = 0;
-      let maxIndex = 0;
+      let maxIndex = undefined;
 
       while (counter < gapLimit) {
         const addr = nextAddrFunc(index);
+        if (addr === undefined) break;
```

Comment:
replace by `!addr`

Reply:
Ok but we create a bug if `addr` is empty string, even though there is no reason to be empty string